godot-macros/src/util/list_parser.rs

```diff
@@ -132,6 +132,20 @@ impl ListParser {
         Ok(Some(kv.ident()?))
     }
 
+    /// Check if the next element of the list is an identifier
+    ///
+    /// Returns `None` if there are no more elements left, `Some(true)` if the
+    /// next element is identifier and `Some(false)` if it is not
+    pub fn is_next_ident(&mut self) -> Option<bool> {
+        let Some(kv) = self.peek() else {
+            return None;
+        };
+
+        let res = kv.as_ident();
+
+        Some(res.is_ok())
+    }
```

Comment:
To have all change requests as suggestions:
* Could be named `peek_next_ident` and return the `Ident` directly.
* Full stops and line lengths in docs 😉

Reply:
I don't really think that `peen_next_ident` will actually work.

The thing is in the `#[export_range(min, max, step, options..)]` the `min` and `max` are the only required elements. Next comes `step`, which (I think?) need to be retrieved using `parser.next_expr()?`. If there were one of the options (which are idents) it will be retrieved as expression - so `parser.next_expr()?` won't throw error, the error would be from general compiler if symbol of the same name as option identifier won't be present in current scope.

I wanted to introduce `parser.try_next_expr()` there, but it wouldn't work because of the above. Instead I needed to do reverse check - if there is identifier, we can omit getting the `step`, and we can move to option parsing.

Later, at the other hand the options are parsed using loop with `parser.next_any_ident(&ALLOWED_OPTIONS[..])?` which is very clear and clean, but unfortunately needs the parser to have all the option identifiers not consumed. If I would retrieve the identifier beforehand, it wouldn't be there. 

**MID-WRITING IDEA**
So, during writing this comment I begin to think in some other way, and came up with an alternative skipping the need for `is_next_ident` :

```rust
        let step = match parser.next_any_ident(&ALLOWED_OPTIONS[..]) {
            Ok(maybe_option) => {
                if let Some(option) = maybe_option {
                    options.insert(option.to_string());
                }
                quote! { None }
            },
            Err(_) => {
                let value = parser.next_expr()?;
                quote!{ Some(#value) }
            }
        };
```
But I'm not completely sure that it will work in all scenarios, seems to pass the tests though...

Reply:
So what do you suggest?

As mentioned, let's just get this working for now, we can refactor the parsers later... Important is that the cases are covered by tests so that any future changes don't break things.

Reply:
Currently the only idea I have is the explicit check for the identifier, as in initial commit with `is_next_ident`, without consuming or retrieving it in any way. It has some overhead and adds next method, needed exclusively for this functionality, but keeps the parsing with explicit and clean error messages.

Could just expose `peek()` method in `ListParser` and move the whole `is_next_ident` method implementation into `new_range_list()`. It should be safe, I think, as `peek()` doesn't mutate the `ListParser`.

Reply:
> Currently the only idea I have is the explicit check for the identifier, as in initial commit with is_next_ident, without consuming or retrieving it in any way. It has some overhead and adds next method, needed exclusively for this functionality, but keeps the parsing with explicit and clean error messages.

That's fine for me 🙂 I'm not too worried about the small overhead at this point, but eventually would like to consolidate the implementation a bit.

So if once you address the other suggestions in the code, it should be ready to merge from my point 👍

Reply:
@Bromeon I think I've adressed all other issues mentioned, it seems to be ready to merge?

Reply:
Almost good, please squash the commits to one. And let me know if you need help with it!

Reply:
@Bromeon I think I've done it correctly this time! I found this strategy: https://stackoverflow.com/a/69827502 to be much clearer to me than venturing into a rebase hellish circle :)